deepspeed/runtime/engine.py

```diff
@@ -2899,6 +2901,20 @@ def _load_zero_checkpoint(self, load_dir, tag, load_optimizer_states=True):
             logger.info(f"loading {len(zero_sd_list)} zero partition checkpoints for rank {self.global_rank}")
         return True
 
+    def update_optimizer_step(self, step):
+        optimizer = self.optimizer
+        base_optimizer = optimizer.optimizer
+        state = base_optimizer.state
+        for group in optimizer.param_groups:
+            if 'step' in group:
+                group['step'] = step
```

Comment:
It might be useful to future-proof this in case `group['step']` ever becomes a tensor, as happened with `state[p]['step']`.

Reply:
will do